src/event/util/log/remove.rs

```diff
@@ -110,8 +124,18 @@ mod test {
         ];
 
         for (query, expected_first, expected_second) in queries.iter() {
-            assert_eq!(remove(&mut fields, query), *expected_first, "{}", query);
-            assert_eq!(remove(&mut fields, query), *expected_second, "{}", query);
+            assert_eq!(
+                remove(&mut fields, query, false),
+                *expected_first,
+                "{}",
+                query
+            );
+            assert_eq!(
+                remove(&mut fields, query, false),
+                *expected_second,
+                "{}",
+                query
+            );
```

Comment:
Would it make sense to add unit tests for cases when the `prune` parameter is `true`?

Reply:
I was figuring that the behavior tests would exercise this path, but a set of unit tests would still make sense. Do you think it is worth having a separate set of no-prune routines that skip all the extra logic in the common case?

Reply:
> Do you think it is worth having a separate set of no-prune routines that skip all the extra logic in the common case?

I don't really have a preference here.

One argument to keep them separate is that the no-prune routine can in principle be reimplemented without recursion later (see https://github.com/timberio/vector/issues/2104), which can improve performance, and the pruning version cannot be implemented without using some form of stack.

However, it is not clear in advance how much performance improvements would the iterative version of the no-prune routine give, so I think we can keep the unified implementation now because it is more DRY and then, if necessary, split it as part of https://github.com/timberio/vector/issues/2104.

Reply:
The other difference is that the no-prune routine doesn't need to handle the tuple return type. Again, we would have to measure to determine if that makes a significant difference.